applications/fishing-map/src/features/app/App.tsx

```diff
@@ -44,6 +53,18 @@ function App(): React.ReactElement {
     dispatch(fetchUserThunk())
   }, [dispatch])
 
+  useEffect(() => {
+    if (userLogged) {
+      if (
+        (locationType === HOME && workspaceStatus !== AsyncReducerStatus.Finished) ||
+        (locationType !== WORKSPACE && currentWorkspaceId !== workspaceId)
+      ) {
+        dispatch(fetchWorkspaceThunk(workspaceId as string))
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userLogged, workspaceId])
```

Comment:
Is possible that `locationType` or `currentWorkspaceId` changes and `workspaceId` stays unmodified? Not sure if it's a valid scenario or not, if so then maybe it would convenient to add them to the array so the `useEffect` hook is executed when they change too.

Reply:
It wasn't possible but as seems to be confusing I've made it more clear, take a look at: https://github.com/GlobalFishingWatch/frontend/commit/cf267d75b9c532e647e96a863d61ab354b2b99e2
Also I got a bug when going to the home from a workspace without the loading finished caused not triggering the default view fetch, it was also fixed and added a validation to avoid multiple fetch.

Reply:
Looks great! 👍